src/lib/containers/SubsectionRowRenderer.tsx

```diff
@@ -0,0 +1,167 @@
+
+const SubsectionRowRenderer: React.FunctionComponent<SubsectionRowRendererProps> = ({
+  sql,
+  searchParams,
+  sqlOperator,
+  isMarkdown = false,
+  columnLink,
+  friendlyValuesMap,
+  columnNameIsSectionTitle = false,
+  limit,
+}) => {
+  const { accessToken } = useSynapseContext()
+  const [rowSet, setRowSet] = useState<RowSet>()
+  const [isLoading, setIsLoading] = useState<boolean>()
+  let mounted = true
+  useEffect(() => {
+    const fetchData = async function () {
+      setIsLoading(true)
+      const sqlUsed = insertConditionsFromSearchParams(
+        sql,
+        searchParams,
+        sqlOperator,
+      )
+      const entityId = parseEntityIdFromSqlStatement(sql)
+      const partMask = SynapseConstants.BUNDLE_MASK_QUERY_RESULTS
+      const request: QueryBundleRequest = {
+        partMask,
+        concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+        entityId,
+        query: {
+          sql: sqlUsed,
+          limit,
+        },
+      }
+
+      const queryResultBundle = await SynapseClient.getFullQueryTableResults(
+        request,
+        accessToken,
+      )
+      setIsLoading(false)
+      const { queryResult } = queryResultBundle
+      const { queryResults } = queryResult
+      if (queryResults) {
+        if (mounted) {
+          setRowSet(queryResults)
+        }
+      } else {
+        console.log('SubsectionRowRenderer: Error getting data')
+      }
+    }
+    fetchData()
+
+    return () => {
+      mounted = false
+    }
+  }, [sql, accessToken, searchParams, sqlOperator])
+
+  /**
+   * If a "friendly values map" was provided, then use the friendly value if any of the raw values match.
+   * Otherwise, just return the raw value.
+   * @param rawValue
+   * @returns 
+   */
+  const getFriendlyValue = (rawValue: string) => {
+    if (!friendlyValuesMap) {
+      return rawValue
+    }
+    const friendlyValue = friendlyValuesMap[rawValue]
+    return friendlyValue ? friendlyValue : rawValue
+  }
+
+  return (
+    <div className="SubsectionRowRenderer bootstrap-4-backport">
+      {isLoading && <SkeletonTable numRows={2} numCols={1} />}
+      {!isLoading && rowSet && rowSet.rows.length > 0 && (
+        rowSet.headers.map((selectColumn, colIndex) => {
+          // If a link column was provided (that contain URLs), do not create a page sub-section for that column.
+          if (columnLink && selectColumn.name == columnLink.linkColumnName) {
+            return <></>
+          }
+          return <div key={`${colIndex}`} className="SubsectionRowRenderer__item" role="table">
+            {/* Is this a page subsection (default), or a page section? */}
+            {!columnNameIsSectionTitle && <h4 className="SubsectionRowRenderer__item__subsection-title" role='heading'>{selectColumn.name}</h4>}
+            {columnNameIsSectionTitle && <><h2 className="SubsectionRowRenderer__item__section-title" role='heading'>{selectColumn.name}</h2><hr /></>}
```

Comment:
Would this be a good place to use the `Typography` component?

Reply:
Good idea!